### PR TITLE
post about atomic system containers and kubernetes

### PR DESCRIPTION
**Pull request by jasonbrooks**

### CODE REVIEW

**Comment by jberkus** [comment]
@bproffitt this looks fine to me, push Monday?

**Comment by cgwalters** [comment]
Can we hold on this a bit and collaborate more on https://github.com/CentOS/sig-atomic-buildscripts/pull/144 ?

**Comment by jberkus** [comment]
@cgwalters why?  This post gives something people can do *right now*.   I can see referencing the ongoing work in the post, though ...

**Comment by jlebon** [comment]
@jberkus Because it's part of a larger story with a better hook. The idea here is that we're trimming the host and migrating to containers. We haven't done the trimming yet (this is what https://github.com/CentOS/sig-atomic-buildscripts/pull/144 does) and the final architecture is still being decided & tested. I'd also be in favour of waiting for that to happen and then rewriting this a bit to get a more compelling story.

**Comment by cgwalters** [comment]
https://lists.projectatomic.io/projectatomic-archives/atomic-devel/2016-August/msg00054.html

**Comment by jasonbrooks** [comment]
This post isn't meant to be about what centos is doing with its alpha branch, it's meant to be about how you can run kube 1.3 on atomic, even though we have no kube 1.3 rpms, and about how to run flannel and etcd in system containers, because that's an interesting new feature that's not really documented yet.

I put it on my personal blog originally because it includes lots of unofficial things, the flannel and etcd containers from Giuseppe and the kube container straight from upstream, and the continuous branch required for the right atomic command version. I can see an argument for putting more fully-baked things on the project atomic blog, and I can see the argument for putting more in-development things on here. 

For instance, we're asking people to +1 the flannel/etcd in system containers idea, but people have had little opportunity to check this out / learn about it to inform their feedback. A post like this gives you a working current kube setup based on these pieces.

Either way, I'm happy to write about the reorganized centos atomic host when it's ready.

**Comment by jberkus** [comment]
@jlebon, @cgwalters :

We support the kubernetes project.  LDK is our stack.  We present stuff at, and sometimes run, Kubernetes meetups.  Kubernetes 1.3 has been out for more than a month.  While our users want to know our roapmap for containerizing all of the services, they also want to be able to run 1.3 on atomic *now*.

So while I think adding a link to the ongoing work makes sense, I disagree that we should wait and not post this because we want to hold off until the system containers are done.  This is an OSS project, we *should* be encouraging people to hack it.

**Comment by cgwalters** [comment]
So now that the CAHC Alpha is out, I think this is also good but would like to see a reference to that as well.

**Comment by jberkus** [comment]
@jasonbrooks do you want to update the post to reference the other one, or do you want me to do it?

**Comment by jasonbrooks** [comment]
@jberkus I'll update it tomorrow

File: source/blog/2016-08-18-running-kubernetes-in-containers-on-atomic.html.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+You need a very recent version of the `atomic` command. I used a pair of CentOS Atomic Hosts running the ["continuous"](https://wiki.centos.org/SpecialInterestGroup/Atomic/Devel) stream.
+
+The master host needs etcd and flannel:
+
+```
+# atomic pull gscrivano/etcd
+
+# atomic pull gscrivano/flannel
+
+# atomic install --system gscrivano/etcd
```
Comment (jlebon) [inline source/blog/2016-08-18-running-kubernetes-in-containers-on-atomic.html.md]: Needs `systemctl start etcd` after this.

File: source/blog/2016-08-18-running-kubernetes-in-containers-on-atomic.html.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+```
+
+With etcd running, we can use it to configure flannel:
+
+```
+# export MASTER_IP=YOUR-MASTER-IP
+
+# runc exec gscrivano-etcd etcdctl set /atomic.io/network/config '{"Network":"172.17.0.0/16"}'
```
Comment (jlebon) [inline source/blog/2016-08-18-running-kubernetes-in-containers-on-atomic.html.md]: s/gscrivano-//

File: source/blog/2016-08-18-running-kubernetes-in-containers-on-atomic.html.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+# atomic install --name=flannel --set ETCD_ENDPOINTS=http://$MASTER_IP:2379 --system gscrivano/flannel
```
Comment (jlebon) [inline source/blog/2016-08-18-running-kubernetes-in-containers-on-atomic.html.md]: systemctl start flannel

File: source/blog/2016-08-18-running-kubernetes-in-containers-on-atomic.html.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+```
+
+The worker node needs flannel as well:
+
+```
+# export MASTER_IP=YOUR-MASTER-IP
+
+# atomic pull gscrivano/flannel
+
+# atomic install --name=flannel --set ETCD_ENDPOINTS=http://$MASTER_IP:2379 --system gscrivano/flannel
```
Comment (jlebon) [inline source/blog/2016-08-18-running-kubernetes-in-containers-on-atomic.html.md]: systemctl start flannel

File: source/blog/2016-08-18-running-kubernetes-in-containers-on-atomic.html.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+```
+# curl -sSL https://storage.googleapis.com/kubernetes-release/release/$(curl -sSL "https://storage.googleapis.com/kubernetes-release/release/stable.txt")/bin/linux/amd64/kubectl > /usr/local/bin/kubectl
+
+# chmod +x /usr/local/bin/kubectl
+```
+
+### test it
+
+It takes a few minutes for all the containers to get up and running. Once they are, you can start running kubernetes apps. I typically test with the [guestbookgo atomicapp](https://github.com/projectatomic/nulecule-library/tree/master/guestbookgo-atomicapp):
```
Comment (jlebon) [inline source/blog/2016-08-18-running-kubernetes-in-containers-on-atomic.html.md]: I would add a `kubectl get nodes` here to help users following along to sanity check their work so far.

**Comment by jlebon** [comment]
This is pretty cool! I made some comments which are mostly to update it to the latest way of doing things in `atomic`.

**Comment by jberkus** [comment]
Looks good.  Barring some kind of objection, I'll push this tommorrow.

**Comment by jberkus** [comment]
Replaced with PR #363 so I could change the date.  Merging that one.

**Pull request closed without merging**